profile-datastore-inmemory/src/main/java/com/telecominfraproject/wlan/profile/datastore/inmemory/ProfileDatastoreInMemory.java

```diff
@@ -46,6 +47,13 @@ public Profile create(Profile profile) {
     		throw new IllegalStateException("unsupported value in profile object");
     	}
     	
+    	for (Profile inMemProfile : idToProfileMap.values()) {
+    		if (profile.getName().equals(inMemProfile.getName()) && 
+    				profile.getProfileType().equals(inMemProfile.getProfileType())) {
+    			throw new DsDuplicateEntityException("Profile with the same name and type already exists");
```

Comment:
Unique combination for the profile attributes is customerId, profileType, and name. You're missing customerId in here.

Reply:
also, need to have the same check in the update() method in this class